### DISCUSSION TITLE
Enhancement request: "Add Data Source" shouldn't automatically save the empty data source

### DISCUSSION THREAD

**Discussion by eamonryan** [grafana/grafana#38248] [Dashboards, Panels & Visualizations]
**What would you like to be added**: When adding a new data source:
- After selecting the data source type
- If no details have been filled in

It would be helpful if the data source was not saved *or* the user was asked whether they wanted to save the data source when navigating away from the page.

Ideally, the data source would not be saved if the tab is force closed or similar. It would only be saved if consciously saved by the user.

**Why is this needed**: Over time, a Grafana installation tends to accumulate a lot of "empty" data sources, from people exploring and seeing what options are needed/available for a specific data source. Despite the "Data source added" green dialog box, I think it's unclear to users that the data source has indeed been saved, even if they haven't entered any information.

I think it would be a more intuitive UX if adding a new data source required explicitly saving the configuration for it to persist and be given a datasource ID.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
